### PR TITLE
Remove more unused attributes in config classes

### PR DESCRIPTION
**Pull request by ydshieh**
# What does this PR do?

### CODE REVIEW

File: src/transformers/models/markuplm/configuration_markuplm.py [hunk 1]
```diff
@@ -65,8 +65,6 @@ class MarkupLMConfig(PretrainedConfig):
             The standard deviation of the truncated_normal_initializer for initializing all weight matrices.
         layer_norm_eps (`float`, *optional*, defaults to 1e-12):
             The epsilon used by the layer normalization layers.
-        gradient_checkpointing (`bool`, *optional*, defaults to `False`):
```
Comment (ydshieh) [inline src/transformers/models/markuplm/configuration_markuplm.py:68]: none of the other config class have `gradient_checkpointing`. In the base class, there is default to `False`

```python
        # Deal with gradient checkpointing
        if kwargs.get("gradient_checkpointing", False):
            warnings.warn(
                "Passing `gradient_checkpointing` to a config initialization is deprecated and will be removed in v5 "
                "Transformers. Using `model.gradient_checkpointing_enable()` instead, or if you are using the "
                "`Trainer` API, pass `gradient_checkpointing=True` in your `TrainingArguments`."
            )
```

File: src/transformers/models/openai/configuration_openai.py [hunk 2]
```diff
@@ -60,8 +60,6 @@ class OpenAIGPTConfig(PretrainedConfig):
             The epsilon to use in the layer normalization layers
         initializer_range (`float`, *optional*, defaults to 0.02):
             The standard deviation of the truncated_normal_initializer for initializing all weight matrices.
-        predict_special_tokens (`bool`, *optional*, defaults to `True`):
```
Comment (ydshieh) [inline src/transformers/models/openai/configuration_openai.py:63]: can't find any usage: not in modeling file, training script, conversion script

File: src/transformers/models/realm/configuration_realm.py [hunk 3]
```diff
@@ -110,8 +110,6 @@ class RealmConfig(PretrainedConfig):
         searcher_beam_size (`int`, *optional*, defaults to 5000):
             Beam size of the searcher. Note that when eval mode is enabled, *searcher_beam_size* will be the same as
             *reader_beam_size*.
-        searcher_seq_len (`int`, *optional*, defaults to 64):
```
Comment (ydshieh) [inline src/transformers/models/realm/configuration_realm.py:113]: Only appear once in testing file

```python
        question = tokenizer(
            [question],
            padding=True,
            truncation=True,
            max_length=model.config.searcher_seq_len,
            return_tensors="pt",
        )
```

File: src/transformers/models/roc_bert/configuration_roc_bert.py [hunk 4]
```diff
@@ -77,8 +77,6 @@ class RoCBertConfig(PretrainedConfig):
             with Better Relative Position Embeddings (Huang et al.)](https://arxiv.org/abs/2009.13658).
         classifier_dropout (`float`, *optional*):
             The dropout ratio for the classification head.
-        enable_cls (`bool`, *optional*, defaults to `True`):
```
Comment (ydshieh) [inline src/transformers/models/roc_bert/configuration_roc_bert.py:80]: no usage at all

File: src/transformers/models/speech_to_text_2/configuration_speech_to_text_2.py [hunk 5]
```diff
@@ -68,8 +68,6 @@ class Speech2Text2Config(PretrainedConfig):
             for more details.
         use_cache (`bool`, *optional*, defaults to `True`):
             Whether or not the model should return the last key/values attentions (not used by all models).
-        max_source_positions (`int`, *optional*, defaults to 6000):
```
Comment (ydshieh) [inline src/transformers/models/speech_to_text_2/configuration_speech_to_text_2.py:71]: `Speech2Text2` only implements decoder models (`Speech2Text2Decoder`, `Speech2Text2DecoderWrapper` and  `Speech2Text2ForCausalLM`), and `max_source_positions` is for `Encoder`, like `Speech2TextEncoder`.

File: src/transformers/models/trajectory_transformer/configuration_trajectory_transformer.py [hunk 6]
```diff
@@ -45,8 +45,6 @@ class TrajectoryTransformerConfig(PretrainedConfig):
         vocab_size (`int`, *optional*, defaults to 100):
             Vocabulary size of the TrajectoryTransformer model. Defines the number of different tokens that can be
             represented by the `trajectories` passed when calling [`TrajectoryTransformerModel`]
-        batch_size (`int`, *optional*, defaults to 256):
```
Comment (ydshieh) [inline src/transformers/models/trajectory_transformer/configuration_trajectory_transformer.py:48]: no usage, and this doesn't fit to a config attribute

File: src/transformers/models/wavlm/configuration_wavlm.py [hunk 7]
```diff
@@ -75,8 +75,6 @@ class WavLMConfig(PretrainedConfig):
         feat_extract_activation (`str, `optional`, defaults to `"gelu"`):
             The non-linear activation function (function or string) in the 1D convolutional layers of the feature
             extractor. If string, `"gelu"`, `"relu"`, `"selu"` and `"gelu_new"` are supported.
-        feat_quantizer_dropout (`float`, *optional*, defaults to 0.0):
```
Comment (ydshieh) [inline src/transformers/models/wavlm/configuration_wavlm.py:78]: other audio models use this attr., but `WavLM` doesn't

File: src/transformers/models/sew_d/configuration_sew_d.py [hunk 8]
```diff
@@ -63,8 +63,6 @@ class SEWDConfig(PretrainedConfig):
             Whether to share attention key with c2p and p2c.
         relative_attention (`bool`, *optional*, defaults to `True`):
             Whether to use relative position encoding.
-        position_biased_input (`bool`, *optional*, defaults to `False`):
```
Comment (ydshieh) [inline src/transformers/models/sew_d/configuration_sew_d.py:66]: This is in the checkpoint conversion script, but never used anywhere else.

It appears for `deberta` and `deberta_v2` where this attribute are actually used

**Review by sgugger — APPROVED** [review]
Thanks for all the fixes!

**Pull request merged**